### ISSUE TITLE
Enhancements and performance improvements

### ISSUE THREAD

**Issue by jcfranco** [neogermi/timezone-grouping#2]
## TL;DR

Made improvements to performance, cross-platform compatibility, and customization options in a fork. Open to collaborate and  contribute these enhancements back to this project.

---

First of all, thanks for this awesome project. Timezone groups is a really interesting and useful concept. I stumbled upon it after reading your awesome [medium](https://bootcamp.uxdesign.cc/the-ux-of-timezones-7476a9685238) [posts](https://s7ng6n.medium.com/how-to-group-time-zones-f8c0f819b6df) and it's clear how much effort and expertise you've put into it.

Secondly, I did some exploration after [forking your repo](https://github.com/jcfranco/timezone-groups) and managed to make the following modifications:

### Enhancements/Improvements

* improved performance
* runs in both Node and in the browser
* accepts arguments to specify:
  * the start date for the time zone group processing
  * the date range for creating the DST-aware time zone groups
  * other date libs to create the time zone groups
  * enable debug mode/messaging
* use `Intl` to provide IANA time zones
* use native APIs as much as possible to avoid using 3rd-party libraries
* simplify output to reduce size and also reference timezone labels by index to allow for dynamic localization
* provide hooks to customize output (experimental and undocumented [your thoughts would be greatly appreciated in this area to ensure the API is simple and flexible])

Would you be open to discussing potential collaboration between our versions? I'd be more than happy to contribute back to your repo if you find any of these changes to be helpful and align with your vision and goals. Your insights would be invaluable.

In any case, I believe these changes could benefit the community, so I've published [`timezone-groups`](https://www.npmjs.com/package/timezone-groups) to make it available for use for now.
